framework/decode/file_processor.cpp

```diff
@@ -306,18 +309,16 @@ bool FileProcessor::ProcessBlocks()
                     // parsed_block, though the block header is still valid.
                     ParsedBlock parsed_block = block_parser.ParseBlock(block_buffer);
 
-                    ProcessVisitor process_visitor(*this);
```

Comment:
I was going to add a comment in the last PR about simplifying this, but vaguely recalled you already had something in mind :)